src/commands/install.ts

```diff
@@ -30,43 +42,92 @@ export default class Install extends Command {
     }),
   };
 
-  private async installSingle(service_id: string) {
-    const service_dir = process.cwd();
-    const parts = service_id.split(':');
-    const service_name = parts[0];
-    let service_tag = parts[1];
-    const config = ServiceConfig.loadFromPath(service_dir);
+  async run() {
+    await this.tasks();
+  }
 
-    if (config.name === service_name) {
-      throw new Error('Services cannot depend on themselves');
-    }
+  async tasks() {
+    const { args, flags } = this.parse(Install);
+    const root_service_path = flags.prefix ? flags.prefix : process.cwd();
+    const dependency_graph = await generateGraphFromPaths([process.cwd()], new EnvironmentConfigV1(), this.app.api);
+    const root_service = Array.from(dependency_graph.nodes.values())[0];
```

Comment:
This won't yield services in the order expected since the graph nodes are a map not an array. This index reference can't be expected to be the "root_service" unless you set `false` as the last param to `generateGraphFromPaths()` which tells it not to recurse through the list.